lib/utils.js

```diff
@@ -187,6 +203,38 @@ var normalizeHintField = function normalizeHintField(hint) {
   return finalHint;
 };
 
+const commandOptions = [
+  'commitQuorum',
+  'writeConcern',
+  'w',
+  'wtimeout',
+  'j',
+  'fsync',
+  'readPreference',
+  'session'
+];
+const keysToOmit = new Set(commandOptions.concat('name', 'key'));
+
+const prepareCreateIndex = function(fieldOrSpec, options) {
```

Comment:
This is a legacy format that attempts to remove issues with shadowing by hoisting functions. It's only really useful if you use `var` and is not generally a concern when we lexically scope our functions. What's kind of worse is that the pattern actually slices type information for e.g. stack traces: we get a readable name `prepareCreateIndex` for us while perusing the source code, but the actual function lacks a name and will show up as anonymous in a stack tace.

Let's just use plain old function definitions in the future: `function prepareCreateIndex(..) {}`

Reply:
I deleted this helper, but will do in the future.